src/easier68k/core/models/list_file.py

```diff
@@ -138,14 +158,93 @@ def load_from_json(self, json_str: str):
         loaded = json.loads(json_str)
         self.symbols = loaded['symbols']
         self.data = loaded['data']
+        self.starting_execution_address = loaded['startingExecutionAddress']
+
+    def read_s_record_filename(self, filepath: str):
+        """
+        Read the S record at the given file path
+        :param filepath: Path to an S record
+        :return:
+        """
+        with open(filepath, 'r') as f:
+            for line in f:
+                # process the line in the file
+                self.__process_s_record_line(line)
+
+    def __process_s_record_line(self, line: str):
+        """
+        Process a single line of the S record
+        This is defined here: http://www.easy68k.com/easy68ksrecord.htm
+        :param line:
+        :return:
+        """
+        # type of record
+        record_type = SRecordType.parse(line[:2])
+        # count of remaining character pairs in the record
+        count = int(line[2:4], 16)
+        # address 2 3 or 4 bytes as hex
+        address_val_len = 0
+
+        if record_type is SRecordType.S0:
+            # address field is unused
+            address_val_len = 4
+        elif record_type is SRecordType.S1:
+            # 2 bytes
+            address_val_len = 4
+        elif record_type is SRecordType.S2:
+            # 3 bytes
+            address_val_len = 6
+        elif record_type is SRecordType.S3:
+            # 4 bytes
+            address_val_len = 8
+        elif record_type is SRecordType.S5:
+            # the address field is interpreted as a 2 byte value
+            # and contains the counts of S1 2 and 3 records prev
+            # transmitted
+            address_val_len = 4
+        elif record_type is SRecordType.S7:
+            # termination record
+            # contains the starting execution address
+            # as 4 bytes
+            address_val_len = 8
+        elif record_type is SRecordType.S8:
+            # termination record
+            # contains the starting execution address
+            # as 3 bytes
+            address_val_len = 6
+        elif record_type is SRecordType.S9:
+            # termination record
+            # contains the starting execution address
+            # as 2 bytes
+            address_val_len = 4

```

Comment:
That's a lot of if elif else going on. Seems overly Is there no way to do a dictionary or something? (I have little experience with the Enum in python)

Reply:
I suppose a dictionary could be used as well, but I figure since there are only 9 modes (and not all of them are used) it shouldn't be a big deal

also leaves it open for handling modes differently